samples/nrf9160/modem_shell/prj.conf

```diff
@@ -79,7 +79,7 @@ CONFIG_NRF_MODEM_LIB_TRACE=n
 
 # AT monitor library
 CONFIG_AT_MONITOR=y
-CONFIG_AT_MONITOR_HEAP_SIZE=512
+CONFIG_AT_MONITOR_HEAP_SIZE=1024
```

Comment:
Good that you pointed this, I had a yellow post-it sticker "Jira ticket" for this in my desk. Not sure about build assert, as this is runtime.
To get forward with my reminder sticker, Just discussed this with modem folks. 512 is enough for legacy types, but with GCI types, NCELLMEAS notification could be much longer. According to modem folks, theoretical max from their side is 4020 bytes!
But what to put here, for now I put just 1k, and eventually that won't be enough in some places.
But putting 4k just for supporting theoretical max for ncellmeas, might be overkill as it consumes usable SRAM from other use cases.
Thus, it was decided that as was done with legacy search types, theoretical max will be tested.

For this PR, I'll put a comment, increase this to 2k and convert my yellow sticker reminder to real Jira ticket and synch with testing.

Reply:
Good point @lemrey, yes, this could be one possible option. But that quite fundamental and I'd left possible change for this for future PRs. We will anyways continue the testings for this, including also with the theoretical max.